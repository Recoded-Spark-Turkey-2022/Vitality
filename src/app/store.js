import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import bookingReducer from '../features/booking/bookingSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    booking: bookingReducer,
  },
});
