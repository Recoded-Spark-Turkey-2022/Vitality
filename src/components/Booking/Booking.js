import React from "react";
// import { Link } from 'react-router-dom';

const Booking = () => {
  return (
    <p>Book an Appointment</p>
  )
}
  export default Booking;

