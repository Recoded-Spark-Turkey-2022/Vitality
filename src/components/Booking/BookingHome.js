import React from 'react';
import { Link } from 'react-router-dom';
import Button from 'react-bootstrap/Button';
import BookingHomeImg from '../../assets/images/BookingHomeImg.png'
import BookingHomeBg from "../../assets/images/BookingHomeBg.png";

const BookingHome = () => {
  return (
    <div style={{ backgroundImage: `url(${BookingHomeBg})`,
    backgroundRepeat: 'no-repeat',
    height: '100vh',
    width: '100vw',
    backgroundSize: 'cover',
    backgroundPosition: 'top',
    top: 0, }}>
      
      <div className="flex flex-col pt-16  items-center md:flex-col   lg:flex-row    lg:justify-evenly lg:items-center md:items-center lg:pt-52  md:pt-28">
        <div className=" flex flex-col items-center   md:flex-col md:items-center  lg:items-center ">
          <p style={{fontFamily: "Poppins",fontWeight: 400,}} className="text-3xl md:text-4xl lg:text-5xl"> WE ARE HERE TO</p>
          <p style={{fontFamily: "Poppins"}} className="text-5xl  md:text-7xl lg:text-9xl ">HELP</p>
        </div>
        
        <div style={{position: "absolute",
width: "300px",
height: "60px",
left: "400px",
top: "675px",}} className="">
          <Link to="/booking">
            <Button variant="info">BOOK AN APPOINTMENT</Button>
          </Link>
        </div>

        <div className="h-2/3 w-2/3 md:h-1/3 md:w-1/3 lg:h-1/3 lg:w-1/3"> 
        <img
                src={BookingHomeImg}
                alt=" "
              />
        </div>
    
    </div>
    </div>
)};

export default BookingHome;
