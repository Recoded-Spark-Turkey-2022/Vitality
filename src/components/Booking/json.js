export const json = { 
    title: "Match with the right therapist.",
    pages: [
      {
        "title": "What is your relationship status?",
        "elements": [
          {
            "type": "checkbox",
            "isRequired": true,
            "choices": [ "Single", "Married", "Divorced" ]
          }
        ]
      },
      {
        "title": "Have you ever been in therapy before?",
        "elements": [
          {
            "type": "checkbox",
            "isRequired": true,
            "choices": [
              "Yes",
              "No"
            ]
          }
        ]
      },
          {
            "type": "radiogroup",
            "title": "Are there any specific qualities that you'd like in a counselor?",
            "choices": [ "I prefer a male counselor", "I prefer a female counselor", "I prefer an older counselor (45+)", "I prefer a non-religious counselor", "I prefer an older counselor (45+)"]
          },
          {
            "type": "radiogroup",
            "title": "Are there any issues you'd like to focus on?",
            "choices": [ "Depression", "Stress and Anxiety", "Relationship issues", "Family conflicts", "Trauma and abuse", "Eating disorders"]
          },
          { type: "comment", title:"What brings you here? Please specify (in a few sentences) why you'd like counseling. This will give your counselor a good understanding of where to start." }
        ]
      };