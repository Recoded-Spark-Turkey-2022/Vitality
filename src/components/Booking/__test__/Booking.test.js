import { Provider } from "react-redux"
import { BrowserRouter } from "react-router-dom"
import renderer from 'react-test-renderer';
import Booking from '../Booking';
import {store} from "../../../app/store"


test('Booking component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store} >
    <BrowserRouter>
        <Booking />
    </BrowserRouter>
</Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});