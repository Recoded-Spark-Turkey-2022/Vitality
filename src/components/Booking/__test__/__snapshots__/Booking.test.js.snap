// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking component renders correctly 1`] = `
<div
  className="w-full md:w-5/6 flex justify-center my-10"
>
  <div
    className="w-5/6 flex flex-col items-center gap-10"
  >
    <div
      className="self-start w-5/6"
    >
      <h3
        className="text-3xl font-semibold"
      >
        LET’S MATCH YOU WITH THE RIGHT THERAPIST
      </h3>
      <p
        className="text-gray-400 text-xl pt-2 font-medium"
      >
        Please fill out this short questionnarie to provide some general and anonymous background about you and the issues you’d like to deal with in online therapy. It would help us match you with the most suitable therapist for you. 
      </p>
    </div>
    <div
      className="border border-solid border-gray-200 h-96 w-full md:w-5/6 lg:w-4/6 px-2 md:px-10 py-5 flex flex-col justify-between items-start rounded-md shadow-xl flex-wrap"
    >
      <div
        className="h-1/6"
      >
        <h2
          className="text-2xl font-medium"
        >
          What Type Of Counseling Are You Looking For?
        </h2>
      </div>
      <div
        className="w-full h-4/6 flex flex-col justify-center md:justify-start gap-2 pt-5"
      >
        <input
          className="w-full border border-solid border-2 border-gray-200 rounded-lg text-md md:text-xl font-semibold pl-2 py-2 text-left hover:bg-gray-100 focus:bg-cyan-400 focus:text-slate-100"
          onClick={[Function]}
          type="button"
          value="Individual Counseling"
        />
        <input
          className="w-full border border-solid border-2 border-gray-200 rounded-lg text-md md:text-xl font-semibold pl-2 py-2 text-left hover:bg-gray-100 focus:bg-cyan-400 focus:text-slate-100"
          onClick={[Function]}
          type="button"
          value="Teen Counseling (For My Child)"
        />
      </div>
      <div
        className="flex justify-start items-center w-5/6 h-1/6"
      >
        <input
          className="py-2 px-5 hover:bg-cyan-200 bg-cyan-400 font-medium rounded"
          onClick={[Function]}
          type="button"
          value="NEXT"
        />
      </div>
    </div>
  </div>
</div>
`;
